src/angular-vs-repeat.js

```diff
@@ -510,11 +510,14 @@
                                 );
                             }
                             else {
-                                __startIndex = Math.max(
-                                    Math.floor(
-                                        ($scrollPosition - $scope.offsetBefore - scrollOffset) / $scope.elementSize
-                                    ) - $scope.excess / 2,
-                                    0
+                                __startIndex = Math.min(
+                                    Math.max(
+                                        Math.floor(
+                                            ($scrollPosition - $scope.offsetBefore - scrollOffset) / $scope.elementSize
+                                        ) - $scope.excess / 2,
+                                        0
+                                    ),
+                                    originalLength
```

Comment:
Although I appreciate the direction of this change, I'm not sure this is safe.

There are many circumstances - especially where using `window` as the scroll-parent - where the expression:

```javascript
($scrollPosition - $scope.offsetBefore - scrollOffset) / $scope.elementSize
```

can become (positive) `Infinity`, but this is a deficiency in the calculation, not an indication that we've actually scrolled well past the end of the list.

In my setup, I have used:

```
vs-repeat
vs-autosize
vs-scroll-parent="window"
```

and this pretty much guarantees that `$scope.elementSize` is zero at all times, which throws the maths out completely.

Ultimately, I think more careful consideration of the maths involved is needed, and the motivation for this fix should be part of it.

Reply:
I have 14 expandable/collapsable vs-repeat containers and 600 items in each. My vs-scroll-parent is a window.
When I expand e.g. 4-th container and scroll down all looks well until I scroll close to the end of the window. At this point my expanded container is not visible in viewport and while I scroll down __startIndex becomes 601, which is wrong. But sometimes __startIndex becomes 602 and this cause digestRequired set to true. After that, digest is triggered and height of the $beforeContent is constantly increasing so I can't scroll to the end at all.